packages/shared/src/hooks/useInView.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState, useRef, useCallback } from 'react';
+import { useIntersectionObserver } from './useIntersectionObserver';
+
+interface UseInViewProps {
+  threshold?: number | number[];
+  root?: Element | null;
+  rootMargin?: string;
+  triggerOnce?: boolean;
+  initialInView?: boolean;
+}
+
+interface UseInViewReturn {
+  ref: (node: HTMLDivElement | null) => void;
+  inView: boolean;
+  entry: IntersectionObserverEntry | null;
+}
+
+export const useInView = ({
```

Comment:
Took implementation from anywhere or? Just to know how in depth to go into this part of code?

Reply:
Nope, so do check please 🙏 I did check what useInView in react-intersection-observer does (what it cares about so we wouldn't lose functionality we need, but didn't add everything)

Reply:
Looks ok to me functionality wise, we need to be careful to test that it works as before since `useLogImpression` relies on it so if anything is off we will screw are metrics pretty good.

Reply:
Yeah, I've tested and looks ok, but still not too confident to merge this today either, so will leave it for Monday.